### ISSUE TITLE
[Bug] Install fails with --namespace helm install parameter if namespace doesn't exist

### ISSUE THREAD

**Issue by cmisale** [IBM/autopilot#48]
# Summary

A `helm install` command with `--namespace=autopilot` will fail if the `autopilot` namespace doesn't exist. This doesn't allow the user to create the namespace automatically through Helm and install the helm chart in that same namespace

## Steps To Reproduce

Made sure autopilot namespace did not exist:

```bash
$ kubectl get namespace autopilot
Error from server (NotFound): namespaces "autopilot" not found
```

Ran command to have the helm release create the namespace for me and targeted the namespace so it was not installed in default:

```bash
$ helm upgrade autopilot autopilot/autopilot --install --create-namespace --namespace=autopilot -f autopilot-cfg.yaml
Release "autopilot" does not exist. Installing it now.
Error: 1 error occurred:
        * namespaces "autopilot" already exists
```

I checked if pods were there anyways, and they were:

```bash
$ kubectl get pods -n autopilot
NAME              READY   STATUS    RESTARTS   AGE
autopilot-2q7zc   0/1     Running   0          26s
autopilot-6s7zz   0/1     Running   0          26s
autopilot-gvprj   0/1     Running   0          26s
autopilot-hbzhl   0/1     Running   0          26s
autopilot-hsfdv   0/1     Running   0          26s
autopilot-hzn7t   0/1     Running   0          26s
autopilot-m87zw   0/1     Running   0          26s
autopilot-n9sbj   0/1     Running   0          26s
autopilot-q9rzb   0/1     Running   0          26s
autopilot-t2z7m   0/1     Running   0          26s
autopilot-zlk99   0/1     Running   0          26s
```

Checked status of autopilot via helm, it was saying it was failed:

```bash
$ helm list 
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS  CHART                   APP VERSION
autopilot       autopilot       1               2024-09-25 17:09:56.366324 -0400 EDT    failed  autopilot-v1.9.0        latest 
```

Ran above command again and it went w/o issues and the helm status changed to deployed...

```bash
$ helm upgrade autopilot autopilot/autopilot --install --create-namespace --namespace=autopilot -f autopilot-cfg.yaml
Release "autopilot" has been upgraded. Happy Helming!
NAME: autopilot
LAST DEPLOYED: Wed Sep 25 17:11:02 2024
NAMESPACE: autopilot
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
Autopilot DaemonSet deployed.

$ helm list                                                                                                          
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
autopilot       autopilot       2               2024-09-25 17:11:02.781032 -0400 EDT    deployed        autopilot-v1.9.0        latest
```

## Expected behavior

Be able to install autopilot in the autopilot namespace

## Evidence

See steps to reproduce

## Proposed Solution

Remove the `create:` option in the `config.yaml` for the namespace, and let it be decided through the Helm cli

**Additional context**
The installer will still work as expected if the helm chart install namespace is different from `autopilot` AND the namespace doesn't exist already.

Thanks to Kevin O'Connor for finding this. All the commands above are reported by him.

**Comment by cmisale** [comment]
I now remember why I did it that way.. [This](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/issues/221 ) shows that Helm doesn't allow to add labels or annotations to the namespace. So it's either we make it helm friendly and add `openshift.io/cluster-monitoring: "true"` through `oc/kubectl`, or make it helm unfriendly and create the namespace with the label. The label is only openshift specific, so taking that out of the way it would make the tool more kubernetes-native..

**Comment by cmisale** [comment]
We decided to remove the namespace creation from the yaml templates and let the user create that through the helm cli.
Further info to enable prometheus scrape metrics from the autopilot repository, will be added in the readme

**Comment by cmisale** [comment]
fixed in #54
